Utilities/XrdAdaptor/src/XrdRequestManager.cc

```diff
@@ -229,12 +230,53 @@ RequestManager::initialize(std::weak_ptr<RequestManager> self)
     m_activeSources.push_back(source);
     updateSiteInfo(orig_site);
   }
+  queueUpdateCurrentServer(source->ID());
+  updateCurrentServer();
 
   m_lastSourceCheck = ts;
   ts.tv_sec += XRD_ADAPTOR_SHORT_OPEN_DELAY;
   m_nextActiveSourceCheck = ts;
 }
 
+/**
+ * Update the StatisticsSenderService with the current server info.
+ *
+ * As this accesses the edm::Service infrastructure, this MUST be called
+ * from an edm-managed thread.  It CANNOT be called from an Xrootd-managed
+ * thread.
+ */
+void
+RequestManager::updateCurrentServer()
+{
+    // NOTE: we use memory_order_relaxed here, meaning that we may actually miss
+    // a pending update.  *However*, since we call this for every read, we'll get it
+    // eventually.
+    if (likely(!m_serverToAdvertise.load(std::memory_order_relaxed))) {return;}
```

Comment:
I don't believe it is worth specifiying the memory order since I've found I've almost always used them incorrectly. Only if it turns out to be a performance problem should we use anything but the default.